source/elements/oneMKL/source/domains/dft/descriptor.rst

```diff
@@ -328,50 +389,49 @@ corresponding configuration values, see :ref:`onemkl_dft_enum_config_param`.
 
 .. container:: section
 
    .. rubric:: Throws
 
-   The `descriptor::get_value()` routine shall throw the following exceptions if the associated condition is detected. An implementation may throw additional implementation-specific exception(s) in case of error conditions not covered here:
+   The ``descriptor::get_value()`` routine shall throw the following exceptions
+   if the associated condition is detected. An implementation may throw
+   additional implementation-specific exception(s) in case of error conditions
+   not covered here:
    
    :ref:`oneapi::mkl::invalid_argument()<onemkl_exception_invalid_argument>`
-      If the requested :ref:`onemkl_dft_enum_config_param` is not correct.
-
-
+      If the requested :ref:`onemkl_dft_enum_config_param` is not valid.
 
 **Descriptor class member table:** :ref:`onemkl_dft_descriptor_member_table`
 
-
-
 .. _onemkl_dft_descriptor_commit:
 
 commit
-+++++++
-
-Finalizes DFT descriptor after all configuration parameters have been set.
-
-.. rubric:: Description
-
-This function completes initialization of a previously created
-descriptor, which is required before the descriptor can be used
-for FFT computations. Typically, committing the
-descriptor performs all initialization that is required for the
-actual FFT computation on the device specified through input queue. 
-The initialization performed by the function may involve exploring different
-factorizations of the input length to find the optimal
-computation method.
-
-All calls to the :ref:`onemkl_dft_descriptor_set_value` function to change configuration
-parameters of a descriptor need to happen after the constructor call for 
-the :ref:`onemkl_dft_descriptor` class and before a call to :ref:`onemkl_dft_descriptor_commit`.
-Typically, a commit function call is immediately followed by a computation 
-function call (see :ref:`onemkl_dft_compute_forward` or :ref:`onemkl_dft_compute_backward`)
-
+++++++
+
+The ``commit`` member function commits a ``descriptor`` object to the DFT
+calculations it defines consistently with its configuration settings, by
+completing all the initialization work (*e.g.*, algorithm selection, algorithm
+tuning, choice of factorization, memory allocations, calculation of twiddle
+factors, etc.) required by the chosen implementation for the desired DFT(s) on
+the targeted device. Objects of the ``descriptor`` class **must** be committed
+prior to using them in any call to :ref:`onemkl_dft_compute_forward` or
+:ref:`onemkl_dft_compute_backward` (which trigger actual DFT calculations).
+
+As specified :ref:`above<onemkl_dft_descriptor_set_value>`, all required
+configuration parameters must be set before this function is called. Any change
+in configuration operated on a ``descriptor`` object via a call to its
+:ref:`onemkl_dft_descriptor_set_value` member function *after* it was committed
+may not be successfully reflected by that object unless its ``commit`` member
+function is called again.
```

Comment:
```suggestion
configuration parameters must be set before this function is called. Any change
in configuration operated on a ``descriptor`` object via a call to its
:ref:`onemkl_dft_descriptor_set_value` member function *after* it was committed
results in an undefined state not suitable for computation until this ``commit`` member
function is called again.
```
I think we should be a little more forceful about disallowing any sort of configuration adjustment without subsequent re-commit.